internal/gatewayapi/translator.go

```diff
@@ -546,11 +548,129 @@ func (t *Translator) ProcessHTTPRoutes(httpRoutes []*v1beta1.HTTPRoute, gateways
 						}
 					}
 
+					// TODO: implement header modifier filter
+					for _, filter := range rule.Filters {
+						if irRoute.DirectResponse != nil {
+							// If an invalid filter type has been configured then an error direct response must be returned
+							// so skip processing any other filters for this route
+							break
+						}
+						switch filter.Type {
+						case "RequestRedirect":
+							// Can't have two redirects for the same route
+							if irRoute.Redirect != nil {
+								parentRef.SetCondition(
+									v1beta1.RouteConditionResolvedRefs,
+									metav1.ConditionFalse,
+									v1beta1.RouteReasonUnsupportedValue,
+									"Cannot configure multiple requestRedirect filters for a single HTTPRouteRule",
+								)
+								continue
+							}
+
+							irRedirect := &ir.Redirect{}
+							if redirect := filter.RequestRedirect; redirect != nil {
+								if redirect.Scheme != nil {
+									// Note that gateway API may support additional schemes in the future, but unknown values
+									// must result in an UnsupportedValue status
+									if *redirect.Scheme == "http" || *redirect.Scheme == "https" {
+										irRedirect.Scheme = redirect.Scheme
+									} else {
+										errMsg := fmt.Sprintf("Scheme: %s is unsupported, only 'https' and 'http' are supported", *redirect.Scheme)
+										parentRef.SetCondition(
+											v1beta1.RouteConditionResolvedRefs,
```

Comment:
I see that the [Gateway API docs](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRequestRedirectFilter) reference setting an "Attached" condition in many of these error cases with a reason "UnsupportedValue", but that condition type doesn't exist for HTTPRoutes. I'm guessing that should actually be the "Accepted" condition, which is listed as having "UnsupportedValue" as a reason. (ref. https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.RouteConditionType).

We should open an issue/contribute a fix to Gateway API for those docs, and assuming that goes through, update these conditions to use "Accepted".

Reply:
I submitted https://github.com/kubernetes-sigs/gateway-api/pull/1382 for this

Reply:
Do you want me to change them all to `Accepted` in this PR? I'd be curious to see what the gateway API folks have to say about it first :thinking: 

I'm also wondering if `Accepted` with a status of `False` is the right move on this since the `HTTPRoute` itself would actually be accepted, just the filter may be invalid/ignored. Happy to change them in this PR if you anticipate the Gateway API folks agreeing.

Reply:
Let's keep as-is for now, I'll create an issue to track the follow-up here, once we see what the Gateway API folks have to say about it. Ultimately we should do whatever is defined in the spec / the conformance tests, I think in this case there was just a typo.

Reply:
https://github.com/envoyproxy/gateway/issues/357